lib/src/view/home/home_tab_screen.dart

```diff
@@ -528,11 +530,12 @@ class _HelloWidget extends ConsumerWidget {
               color: context.lichessColors.brag,
             ),
             const SizedBox(width: 5.0),
-            Text(
-              'Hello${user != null ? ', ' : ''}',
-              style: style,
-            ),
-            if (user != null) UserFullNameWidget(user: user, style: style),
+            if (user != null && greetingParts.length == 2) ...[
```

Comment:
Not sure about this. Is the `greetingParts.length` always 2?

Reply:
Yes, according to the flutter docs:

`A non-empty match at the start or end of the string, or after another match, is not treated specially, and will introduce empty substrings in the result`

So we'll always get 2 here, as long as '%s' is somewhere in the string, including start and end. (If not, it's a translation issue, the comment says to always include the %s)

Reply:
I think it's a common use case (inclusion of a widget instead of string in a translation) so it should be factored out in a utility function.

And this function should have unit tests.

Reply:
Ah yes good point, will do 👍

Reply:
Ok, added a utility function

```
IList<Widget> replaceL10nPlaceholderWithWidget<T extends Widget>(
  String Function(String) l10nWithReplacement,
  T widget, {
  TextStyle? textStyle,
})
```
To avoid empty text widgets, this now also catches the case where the placeholder is at start the start or end and only adds one instead of two Text Widgets in that case